pkg/services/query/query.go

```diff
@@ -90,18 +90,41 @@ func (s *Service) QueryData(ctx context.Context, user *user.SignedInUser, skipCa
 	g, ctx := errgroup.WithContext(ctx)
 	results := make([]backend.Responses, len(byDataSource))
 
+	// Create panic recovery function for loop below
+	recoveryFn := func(results *[]backend.Responses, queries []*simplejson.Json) {
+		if r := recover(); r != nil {
+			var err error
+			s.log.Error("query datasource panic", "error", r, "stack", log.Stack(1))
+			if theErr, ok := r.(error); ok {
+				err = theErr
+			} else if theErrString, ok := r.(string); ok {
+				err = fmt.Errorf(theErrString)
+			} else {
+				err = fmt.Errorf("unexpected error, see the server log for details")
+			}
+			// Because of the panic, there is no valid response for any query. Append an error for each one.
+			*results = append(*results, buildErrorResponses(err, queries))
```

Comment:
I'm not sure this statement holds true since queries are executing in different goroutines. Nevertheless, we should support one datasource panic and another datasource returning actual results. Same potential problem with multiple goroutines writing to `results`.